File: ios/bridge/HybridNitroPlayerViewManagerFactory.swift
//
//  HybridNitroPlayerViewManagerFactory.swift
//  NitroPlay
//
//  Created by Krzysztof Moch on 23/09/2024.
//

import Foundation

class HybridNitroPlayerViewManagerFactory: HybridNitroPlayerViewManagerFactorySpec {
  func createViewManager(nitroId: Double) throws -> any HybridNitroPlayerViewManagerSpec {
    return try HybridNitroPlayerViewManager(nitroId: nitroId)
  }
}

File: android/src/main/java/com/nitroplay/bridge/HybridNitroPlayerViewManagerFactory.kt
package com.margelo.nitro.video

import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
class HybridNitroPlayerViewManagerFactory: HybridNitroPlayerViewManagerFactorySpec() {
  override fun createViewManager(nitroId: Double): HybridNitroPlayerViewManagerSpec {
    return HybridNitroPlayerViewManager(nitroId.toInt())
  }

  override val memorySize: Long
    get() = 0
}
